backend/mem/file.go

```diff
@@ -0,0 +1,219 @@
+package mem
+
+import (
+	"bytes"
+	"errors"
+	"github.com/c2fo/vfs/v4"
+	"io"
+	"time"
+)
+
+//File implements vfs.File interface for os fs.
+type File struct {
+	exists 		bool
+	timeStamp 	time.Time
+	isOpen		bool
+	isZB		bool
+	isRef		bool
+	privSlice 	[] byte
+	byteBuf  *bytes.Buffer
+	Filename string
+	cursor 		int
+	location vfs.Location
+}
+
+func DoesNotExist() error {
+	return errors.New("This file does not exist!")
+}
+func CopyFail() error {
+	return errors.New("This file was not successfully copied")
+}
+func DeleteError() error {
+	return errors.New("Deletion was unsuccessful")
+}
+
+
+func (f *File) Close() error {  //NOT DONE
+	if !f.isRef {
+		// Do nothing on files that were never referenced
+		return nil
+	}
+	if f.byteBuf.Len() > 0{
+		f.privSlice = append(f.privSlice,f.byteBuf.Next(200)...) //TODO: maybe change the number for "Next" arg
+	}
+
+	f.isOpen =false
```

Comment:
There are some spacing issues in this file that would (I think) be fixed by running go-fmt

Reply:
@jkc `gofmt` doesn't fix the problems with empty lines. It will remove the whitespace in an empty line, like `s/\s*$//g` but not the problems with empty lines after function signatures.

I wrote this sed script that fixes them: 

``` sh
sed -e '1h;2,$H;$!d;g' -e 's|{\n\s*|{|g;s|\s*\n}|}|g'
```

Reply:
I was in part referring to the extra spacing throughout the file, but in this particular comment I was mostly pointing to the lack of a space before 'false' ie: '=false' should be '= false'